### DISCUSSION TITLE
Changing the styling format of jsPsych survey likert plugin

### DISCUSSION THREAD

**Discussion by sankalp35** [jspsych/jsPsych#1917] [Q&A]
Hello all,

I wonder if there is any easy way to change the formatting of the survey likert plugin, so that it looks more tabular (see image). It seems like in the current version, participants have to go through many repeat versions of the likert scale, whereas it stays the same for most of my questionnaires.

![image](https://user-images.githubusercontent.com/7095969/122787709-8192f880-d2ad-11eb-886d-e34c63542f0e.png)

**Comment by becky-gilbert** [comment]
Hi @sankalp35, I might be wrong, but I don't think there's an easy way to do this right now. However we are planning to do a major improvement to the survey-* plugins soon (see Issue #1207), and I think this is a good formatting option to provide. So I'll make a note about it on the Issue thread.

In the meantime, you can do this by editing the survey-likert plugin, specifically the way that the page HTML is created, which is around these lines: https://github.com/jspsych/jsPsych/blob/ae332b33477d686388bafbd03b0c8e1c9945999c/plugins/jspsych-survey-likert.js#L126-L142

This code loops over the questions, and for each one, adds the question prompt and scale options to the page HTML. Here, instead of using an unordered list element (`<ul>`), with each option as a list item (`<li>`), you could create a table element with one question per row. The first cell in each row would contain the prompt, and the following cells would have each of the scale options. [This page](https://www.w3schools.com/html/html_tables.asp) has a nice overview of creating HTML tables. Feel free to post your code here if you need further guidance, and we'll help out if we can!

**Comment by sankalp35** [comment]
Thanks Becky, I was able to create this:
![image](https://user-images.githubusercontent.com/7095969/123262685-12591680-d4f0-11eb-9b55-9bd97516d83c.png)

It doesn't look as well on laptop screens so I decided to go ahead with the original version, but it was good to understand some underlying code of the plugin and Its styling.

**Comment by sankalp35** [comment]
![Screenshot 2021-06-24 at 13 31 09](https://user-images.githubusercontent.com/7095969/123263049-75e34400-d4f0-11eb-8c42-f2f2afa650c7.png)

Another quick question about the styling of this plugin: how do I fix this (see image above). The text appears too close to each other in the options. I tried adding a padding-left to some of this code, but none of it worked

```    html += '<style id="jspsych-survey-likert-css">';
    html += ".jspsych-survey-likert-statement { display:block; font-size: 16px; padding-top: 40px; margin-bottom:10px; }"+
      ".jspsych-survey-likert-opts { list-style:none; width:"+w+"; margin:auto; padding:0 0 35px; display:block; font-size: 14px; line-height:1.1em; }"+
      ".jspsych-survey-likert-opt-label { line-height: 1.1em; color: #444; }"+
      ".jspsych-survey-likert-opts:before { content: ''; position:relative; top:11px; /*left:9.5%;*/ display:block; background-color:#efefef; height:4px; width:100%; }"+
      ".jspsych-survey-likert-opts:last-of-type { border-bottom: 0; }"+
      ".jspsych-survey-likert-opts li { display:inline-block; /*width:19%;*/ text-align:center; vertical-align: top; }"+
      ".jspsych-survey-likert-opts li input[type=radio] { display:block; position:relative; top:0; left:50%; margin-left:-6px; }"
    html += '</style>';
```

**Reply by becky-gilbert** [reply]
Hi @sankalp35, can you increase the scale width? Other than that, I couldn't think of a way to do this without editing the plugin file.

If you're happy to modify the plugin, then one solution is to put the scale option label inside of a `div` element, and then reduce the width of that `div` to e.g. 80%. The HTML for the option labels is set up in [line 138 of the v6.3.1 plugin file](https://github.com/jspsych/jsPsych/blob/7b16a1d39ab38b3a01c33c9d0a2fec06524e81ee/plugins/jspsych-survey-likert.js#L138). You can add a `div` around the label and reduce its width like this:
```js
options_string += '><div style="width:80%;">' + question.labels[j] + '</div></label></li>';
```
Does that solution work for you?

**Reply by sankalp35** [reply]
Thanks Becky, I will be doing this for the next version of my task. I'll get back to you!

**Reply by acer-helia** [reply]
A really sneaky option for changing the text if you don't want to change the scale width/plugin is to write in custom breaks when you're specifying your scale labels, as below:

```javascript
var scale_fatigue = [
      "<p>I am not at all<br> fatigued</p>",
      "<p>I am a little<br> bit fatigued</p>",
      "<p>I am feeling<br> moderately<br> fatigued</p>",
      "<p>I am too tired<br> and fatigued to do<br> a lot of the things<br>I used to do</p>",
      "<p>I am too tired<br> and fatigued to do<br> most of the things<br>I used to do</p>",
    ]
```

The only caution is that the breaks may be different across different types of presentations (e.g., desktop vs phone) but it's been a pretty good option across browsers for me.

**Reply by sankalp35** [reply]
Hi @acer-helia, 
That was so sneaky and so clever. Worked like magic, and without the hassle of digging into the plugin! Thanks
